app.py

```diff
@@ -0,0 +1,97 @@
+import panel as pn
+import hvplot.pandas
+
+# Load Data
+from scripts.data import per_dataset_df, all_framework_df
+from scripts.widgets import create_selectwidget, create_togglewidget, create_numberwidget
+from scripts.plots import create_hvplot,create_table
+
+#clean up framework names
+original_framework_names = all_framework_df['framework']
+framework_names = per_dataset_df['framework'].str.extract(r"^(.*?)(?:_|$)")[0]
+per_dataset_df.framework = framework_names
+all_framework_df.framework = framework_names
+
+dataset_list = sorted(list(set(per_dataset_df['dataset'].to_list())))
+
+# Make DataFrame Pipeline Interactive
+per_dataset_idf = per_dataset_df.interactive()
+all_framework_idf = all_framework_df.interactive()
+
+# Define Panel widgets
+metrics = ['loss_rescaled', 'time_train_s_rescaled','time_infer_s_rescaled', 'bestdiff']
+graph_types = ['line', 'bar','hist', 'scatter']
+frameworks_widget = create_selectwidget('Framework', options=framework_names.to_list())
+yaxis_widget =  create_selectwidget('Y-axis Metric', options=metrics)
+yaxis_widget2 =  create_selectwidget('Y-axis Metric', options=metrics)
+datasets_widget = create_togglewidget('Datasets', dataset_list)
+dataset_dropdown = create_selectwidget('Datasets', options=dataset_list)
+graph_type = create_selectwidget('Graph Type', 'bar', graph_types)
+graph_type2 = create_selectwidget('Graph Type', 'bar', graph_types)
+
+#Some data processing
+idf_dataset = per_dataset_idf[(per_dataset_idf.dataset.isin([dataset_dropdown]))]
+df_ag_only = per_dataset_df.loc[(per_dataset_df['framework'].isin(['AutoGluon']))]
+df_ag_only = df_ag_only.sort_values(by=['rank'])
+prop_ag_best = df_ag_only.loc[(per_dataset_df['rank'] == 1.0)].shape[0] / len(dataset_list)
+#prop_ag_best = all_framework_df[all_framework_df['framework'] == 'AutoGluon']['winrate'][0]
```

Comment:
remove if not needed

Reply:
I have added it as an alternative way to get the proportion of 1st ranks for AutoGluon. The uncommented method is by looking at the number of rows where autogluon has rank 1.0 and then divides by number of datasets. The commented method is reading directly from the all datasets combined csv. It just depends on how our final csv's are formatted and what the eval logic is doing.